multi-agent/adapters/inbound/flask/endpoints/sessions.py

```
@@ -246,12 +247,72 @@ def get_session_status(session_id):
+    """Parse and allowlist the ``filters`` query param.
+
+    Returns the parsed dict, or raises ``ValueError`` with a client-facing
+    message if the value is malformed or contains unsupported keys.
+    """
+    if not raw_filters:
+        return {}
+
+    try:
+        parsed = json.loads(raw_filters)
+    except (json.JSONDecodeError, TypeError):
+        raise ValueError("filters must be a valid JSON object")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)